internal/receiver/smartagentreceiver/receiver_test.go

```diff
@@ -242,3 +264,100 @@ func TestConfirmStartingReceiverWithInvalidMonitorInstancesDoesntPanic(t *testin
+	cfg := newConfig("valid", "cpu", 1)
+	observedLogger, logs := observer.New(zapcore.WarnLevel)
+	logger := zap.New(observedLogger)
+	r := NewReceiver(logger, cfg, consumertest.NewMetricsNop())
+	configs := getSmartAgentExtensionConfig(t)
+	host := &mockHost{
+		smartagentextensionConfig:      configs[0],
+		smartagentextensionConfigExtra: configs[1],
+	}
+
+	require.NoError(t, r.Start(context.Background(), host))
+	require.NoError(t, r.Shutdown(context.Background()))
+	require.Equal(t, 1, logs.Len())
+	require.Equal(t, "multiple smartagent extensions found, using the first one encountered", logs.All()[0].Message)
+	require.Equal(t, &config.CollectdConfig{
+		DisableCollectd:      false,
+		Timeout:              10,
+		ReadThreads:          1,
+		WriteThreads:         4,
+		WriteQueueLimitHigh:  5,
+		WriteQueueLimitLow:   400000,
+		LogLevel:             "notice",
+		IntervalSeconds:      10,
+		WriteServerIPAddr:    "127.9.8.7",
+		WriteServerPort:      0,
+		ConfigDir:            "/etc/",
+		BundleDir:            "/opt/",
+		HasGenericJMXMonitor: true,
+		InstanceName:         "",
+		WriteServerQuery:     "",
+	}, saConfigProvider.CollectdConfig())
+
+	// Ensure envs are setup.
+	require.Equal(t, "/opt/", os.Getenv("SIGNALFX_BUNDLE_DIR"))
+	require.Equal(t, filepath.Join("/opt", "jre"), os.Getenv("JAVA_HOME"))
+}
+
+func getSmartAgentExtensionConfig(t *testing.T) []*smartagentextension.Config {
+	factories, err := componenttest.ExampleComponents()
+	require.Nil(t, err)
+
+	factory := smartagentextension.NewFactory()
+	factories.Extensions[typeStr] = factory
+	cfg, err := configtest.LoadConfigFile(
+		t, path.Join(".", "testdata", "extension_config.yaml"), factories,
+	)
+	require.NoError(t, err)
+
+	partialSettingsConfig := cfg.Extensions["smartagent/partial_settings"]
+	require.NotNil(t, partialSettingsConfig)
+
+	extraSettingsConfig := cfg.Extensions["smartagent/extra"]
+	require.NotNil(t, extraSettingsConfig)
+
+	one, ok := partialSettingsConfig.(*smartagentextension.Config)
+	require.True(t, ok)
+
+	two, ok := extraSettingsConfig.(*smartagentextension.Config)
+	require.True(t, ok)
+	return []*smartagentextension.Config{one, two}
+}
+
+type mockHost struct {
+	smartagentextensionConfig      *smartagentextension.Config
+	smartagentextensionConfigExtra *smartagentextension.Config
+}
+
+func (m *mockHost) ReportFatalError(err error) {
+}
+
+func (m *mockHost) GetFactory(kind component.Kind, componentType configmodels.Type) component.Factory {
+	return nil
+}
+
+func (m *mockHost) GetExtensions() map[configmodels.Extension]component.ServiceExtension {
+	randomExtensionConfig := &healthcheckextension.Config{}
+	return map[configmodels.Extension]component.ServiceExtension{
+		m.smartagentextensionConfig:      getExtension(smartagentextension.NewFactory(), m.smartagentextensionConfig),
+		randomExtensionConfig:            getExtension(healthcheckextension.NewFactory(), randomExtensionConfig),
+		m.smartagentextensionConfigExtra: nil,
```

Comment:
Curious why this is nil and not using a created extension from `smartagentextensionConfigExtra`?

Reply:
The receiver uses only the config of the extension, so we don't really about the actual extension itself.